odoo/osv/expression.py

```diff
@@ -329,6 +329,83 @@ def distribute_not(domain):
 
     return result
 
+def remove_domain_leaf(domain, fields_to_remove):
+    if len(domain) < 1:
+        return []
+    # Each character in the transformation_patern dict represent an element in the domain:
+    #       -'0' represents a tuple to remove from the domain
+    #       -'1' represents a tuple to keep in the domain
+    #       -'2' represents a logical operator
+    #       -'3' represents a '!' (not)
+    # e.g. the domain/part of a domain ['|', (tuple to remove), (tuple to keep)] is represented by '201'
+    # and will be replaced by '.1', i.e. [(tuple to keep)].
```

Comment:
replaced by '..1'

Reply:
Thank you for your comment :+1: The first version of the algorithm was not really satisfying in terms of clarity and code complexity. I've updated this PR with a new version of the function ;)